packages/@n8n/decorators/src/module.ts

```diff
@@ -21,6 +27,17 @@ export const N8nModule = (): ClassDecorator => (target) => {
 	return Service()(target);
 };
 
+interface ExecutionLifecycleHooks {
+	addHandler<
+		Hook extends
+			| 'nodeExecuteBefore'
+			| 'nodeExecuteAfter'
+			| 'workflowExecuteBefore'
+			| 'workflowExecuteAfter',
+		// eslint-disable-next-line @typescript-eslint/ban-types
+	>(hookName: Hook, ...handlers: Function[]): void;
+}
+
 @Service()
 export class ModuleRegistry {
```

Comment:
should we move `ModuleRegistry` into `cli`, and keep this package only for the metadata stuff?